Gems/AWSCore/Code/Tools/ResourceMappingTool/utils/json_utils.py

```diff
@@ -119,6 +130,24 @@ def convert_json_dict_to_resources(json_dict: Dict[str, any]) -> List[ResourceMa
     return resources
 
 
+def load_resource_mapping_json_schema(schema_path: str) -> None:
+    global _RESOURCE_MAPPING_SCHEMA, _RESOURCE_MAPPING_SCHEMA_REQUIRED_ROOT_PROPERTIES, _RESOURCE_MAPPING_SCHEMA_REQUIRED_RESOURCE_PROPERTIES,\
+        _RESOURCE_MAPPING_SCHEMA_ACCOUNTID_PATTERN, _RESOURCE_MAPPING_SCHEMA_REGION_PATTERN, _RESOURCE_MAPPING_SCHEMA_VERSION_PATTERN
+    if not _RESOURCE_MAPPING_SCHEMA:
+        # assume schema should be in correct format, and manually load expected pattern; tool will log error if schema is invalid
+        _RESOURCE_MAPPING_SCHEMA = read_from_json_file(schema_path)
+        _RESOURCE_MAPPING_SCHEMA_REQUIRED_ROOT_PROPERTIES = _RESOURCE_MAPPING_SCHEMA[_RESOURCE_MAPPING_SCHEMA_REQUIRED_PROPERTIES_KEY_NAME]
+        schema_pattern_properties: Dict[str, any] = \
+            _RESOURCE_MAPPING_SCHEMA[_RESOURCE_MAPPING_SCHEMA_PROPERTIES_KEY_NAME][_RESOURCE_MAPPING_JSON_KEY_NAME][_RESOURCE_MAPPING_SCHEMA_PATTERN_PROPERTIES_KEY_NAME]
+        _RESOURCE_MAPPING_SCHEMA_REQUIRED_RESOURCE_PROPERTIES = list(schema_pattern_properties.values())[0][_RESOURCE_MAPPING_SCHEMA_REQUIRED_PROPERTIES_KEY_NAME]
+        _RESOURCE_MAPPING_SCHEMA_ACCOUNTID_PATTERN = \
+            _RESOURCE_MAPPING_SCHEMA[_RESOURCE_MAPPING_SCHEMA_ACCOUNTID_KEY_NAME][_RESOURCE_MAPPING_SCHEMA_PROPERTY_PATTERN_KEY_NAME]
+        _RESOURCE_MAPPING_SCHEMA_REGION_PATTERN = \
+            _RESOURCE_MAPPING_SCHEMA[_RESOURCE_MAPPING_SCHEMA_REGION_KEY_NAME][_RESOURCE_MAPPING_SCHEMA_PROPERTY_PATTERN_KEY_NAME]
+        _RESOURCE_MAPPING_SCHEMA_VERSION_PATTERN = \
+            _RESOURCE_MAPPING_SCHEMA[_RESOURCE_MAPPING_SCHEMA_PROPERTIES_KEY_NAME][_RESOURCE_MAPPING_VERSION_JSON_KEY_NAME][_RESOURCE_MAPPING_SCHEMA_PROPERTY_PATTERN_KEY_NAME]
```

Comment:
some of these value lookups are quite nested and hard to read. What do you think about extracting portions of the schema out as local vars to make it easier to understand? We could at least do this with "properties", since we query it more than once:

```py
# extract "properties" dict
schema_properties: Dict[str, any] = _RESOURCE_MAPPING_SCHEMA[_RESOURCE_MAPPING_SCHEMA_PROPERTIES_KEY_NAME]

schema_pattern_properties: Dict[str, any] = \
	schema_properties[_RESOURCE_MAPPING_JSON_KEY_NAME][_RESOURCE_MAPPING_SCHEMA_PATTERN_PROPERTIES_KEY_NAME]
...
_RESOURCE_MAPPING_SCHEMA_VERSION_PATTERN = \
	schema_properties[_RESOURCE_MAPPING_VERSION_JSON_KEY_NAME][_RESOURCE_MAPPING_SCHEMA_PROPERTY_PATTERN_KEY_NAME]
```

Reply:
will update